Update README.md (#102)
xebia-functional/books-store-training: README.md

## Run

### Docker

> Creation of a file named "docker-compose.yml" at root level with the following code:
> 
> ```
> version: '3.8'
> 
> services:
>   db:
>     image: postgres:15
>     environment:
>       POSTGRES_USER: ${POSTGRES_USER:-xebia_user}
>       POSTGRES_PASSWORD: ${POSTGRES_PASSWORD:-xebia_pw}
>       POSTGRES_DB: ${POSTGRES_DB:-bookstore_db}
>     ports:
>       - "5432:5432"
> 
>     healthcheck:
>       test: [ "CMD", "pg_isready", "-U", "${POSTGRES_USER:-xebia_user}" ]
>       interval: 10s
>       start_period: 5s
> ```

Reviewer: ```suggestion
Creation of a file named [docker-compose.yml](docker-compose.yml) for the database.
```

### Flyway

> Creation of flyway (src/main/java/com/xebia/migrations) for the migrations
> 
> ```
> public class FlywayMigration {
>   public static void main(String[] args) {}
> 
>   public static void migrate(String url, String username, String password) {
>     Flyway flyway =
>         Flyway.configure()
>             .dataSource(url, username, password)
>             .locations("classpath:db/migrations")
>             .load();
> 
>     flyway.migrate();
>   }
> }
> ```

Reviewer: ```suggestion
Integration of Flyway for the migrations. The migrations are stored here [src/main/resources/db/migrations](src/main/resources/db/migrations).
```

## Libraries
Libraries are published in Maven Central:
- JUnit for memory tests
- Flyway for migrations
- postgres for the database
- Testcontainers for database test

You may need to add that repository explicitly in your build, if you haven't done it before.

```
repositories { mavenCentral() }
```
Then add the libraries in the usual way.

```

[libraries]
junit-jupiter-bom = { module = "org.junit:junit-bom", version.ref = "junit" }
flyway = { module = "org.flywaydb:flyway-core", version.ref = "flyway" }
postgresql= {module = "org.postgresql:postgresql", version.ref = "postgre"}
testcontatiners = {module = "org.testcontainers:postgresql", version.ref = "testcontainers"}

```

Then add the respective dependencies
```
dependencies {
    testImplementation(platform(libs.junit.jupiter.bom))
    testImplementation("org.junit.jupiter:junit-jupiter")
    implementation(libs.flyway)
    implementation(libs.postgresql)
    testImplementation(libs.testcontatiners)
}

> ```

Reviewer: Usually it's better not to put the code itself as it is defined in the files. It could be better to just refer to the files itself.
